pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -417,6 +417,14 @@ stdenv.mkDerivation ({
   configurePlatforms = [];
   inherit configureFlags;
 
+  # Note: the options here must be always added, regardless of whether the
+  # package specifies `hardeningDisable`.
+  hardeningDisable = lib.optionals (args ? hardeningDisable) hardeningDisable
+    ++ lib.optional (ghc.isHaLVM or false) "all"
+    # Static libraries fails to build if Nix forcibly adds `-pie` to the linker
+    # flags: it conflicts with `-r` and `-no-pie` added by GHC (see ghc/ghc#19580).
```

Comment:
Can you turn `ghc/ghc#19580` into an actual link to the GHC issue tracker?

------------------------------------------------------

Also, I don't quite understand the combination of the comment here with the code.

The comment says: `Static libraries fails to build if Nix forcibly adds `-pie` to the linker flags`.  But then the code does appear to add `pie` to `hardeningDisable`?

Reply:
It seems that by default Nix adds `-pie` to the linker flags, while GHC adds `-r -no-pie` which have the opposite meaning and so the linking fails. Adding `hardeningDisable = [ "pie" ]` prevent Nix from adding `-pie`: the option has a negative meaning.

Reply:
The comment should be a bit clearer, now.